src/transactionIntentsFactories/delegationTransactionIntentsFactory.ts

```diff
@@ -0,0 +1,344 @@
+import { IAddress } from "../interface";
+import { BigNumber } from "bignumber.js";
+import { numberToPaddedHex, byteArrayToHex, utf8ToHex } from "../utils.codec";
+import { TransactionIntent } from "../transactionIntent";
+import { TransactionIntentBuilder } from "./transactionIntentBuilder";
+import { Address } from "../address";
+import { DELEGATION_MANAGER_SC_ADDRESS } from "../constants";
+import { Err } from "../errors";
+
+interface Config {
+    chainID: string;
+    minGasLimit: BigNumber.Value;
+    gasLimitPerByte: BigNumber.Value;
+    gasLimitStake: BigNumber.Value;
+    gasLimitUnstake: BigNumber.Value;
+    gasLimitUnbond: BigNumber.Value;
+    gasLimitCreateDelegationContract: BigNumber.Value;
+    gasLimitDelegationOperations: BigNumber.Value;
+    additionalGasLimitPerValidatorNode: BigNumber.Value;
+    additionalGasLimitForDelegationOperations: BigNumber.Value;
+}
+
+interface ValidatorPublicKey {
+    hex(): string;
+}
+
+export class DelegationTransactionIntentsFactory {
+    private readonly config: Config;
+
+    constructor(config: Config) {
+        this.config = config;
+    }
+
+    createTransactionIntentForNewDelegationContract(options: {
+        sender: IAddress,
+        totalDelegationCap: BigNumber.Value,
+        serviceFee: BigNumber.Value,
+        value: BigNumber.Value
+    }): TransactionIntent {
+        const dataParts = [
+            "createNewDelegationContract",
+            numberToPaddedHex(options.totalDelegationCap.toString()),
+            numberToPaddedHex(options.serviceFee.toString())
+        ];
+
+        const executionGasLimit = new BigNumber(this.config.gasLimitCreateDelegationContract).plus(this.config.additionalGasLimitForDelegationOperations);
+
+        return new TransactionIntentBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: Address.fromBech32(DELEGATION_MANAGER_SC_ADDRESS),
+            dataParts: dataParts,
+            executionGasLimit: executionGasLimit,
+            value: options.value
+        }).build();
+    }
+
+    createTransactionIntentForAddingNodes(options: {
+        sender: IAddress,
+        delegationContract: IAddress,
+        publicKeys: ValidatorPublicKey[],
+        signedMessages: Uint8Array[]
+    }): TransactionIntent {
+        if (options.publicKeys.length !== options.signedMessages.length) {
+            throw new Err("The number of public keys should match the number of signed messages");
+        }
+
+        const numNodes = options.publicKeys.length;
+
+        let dataParts = ["addNodes"];
+        for (let i = 0; i < numNodes; i++) {
+            dataParts = dataParts.concat(options.publicKeys[i].hex());
+            dataParts = dataParts.concat(byteArrayToHex(options.signedMessages[i]));
+        }
+
```

Comment:
If possible, we can have `const dataParts`, and use [.push()](https://stackoverflow.com/a/35167699).